integrations/flink_connector/sample-kinesis-to-timestream-app/src/main/java/com/amazonaws/samples/kinesis2timestream/TimestreamInitializer.java

```diff
@@ -15,31 +15,23 @@
  * Checks if required database and table exists in Timestream. If they do not exists, it creates them
  */
 public class TimestreamInitializer {
-    private final TimestreamWriteClient writeClient;
+    private final String region;
+    private final String endpointOverride;
 
     public TimestreamInitializer(String region, String endpointOverride) {
-        if (endpointOverride != null) {
-            URI endpointOverrideURI = parseEndpointOverride(endpointOverride);
-            this.writeClient = TimestreamWriteClient
-                    .builder()
-                    .region(Region.of(region))
-                    .endpointOverride(endpointOverrideURI)
-                    .build();
-        } else {
-            this.writeClient = TimestreamWriteClient
-                    .builder()
-                    .region(Region.of(region))
-                    .build();
-        }
+        this.region = region;
+        this.endpointOverride = endpointOverride;
     }
 
     public void createDatabase(String databaseName) {
```

Comment:
Would it be better to refactor it into 2 methods - createDatabase and createTable?